VotingApplication/VotingApplication.Web.Api/Scripts/Signin.js

```diff
@@ -0,0 +1,94 @@
+﻿define(['jquery', 'knockout', 'Navbar'], function ($, ko, Navbar) {
+    return function SigninViewModel() {
+        var self = this;
+
+        self.username = ko.observable("");
+        self.password = ko.observable("");
+
+        self.newUsername = ko.observable("");
+        self.newPassword = ko.observable("");
+        self.confirmPassword = ko.observable("");
+
+        var checkPasswordFormat = function (password) {
+            // Checks for password with uppercase, lowercase, digits and punctuation.
+            // AccountController fails the password if any of these are missing
+            var passwordRegex = /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[!@#$&*]).*$/;
```

Comment:
Personally, I hate it when websites enforce this. Can we not instead enforce a lower limit on password length?